WinUIGallery/DataModel/ControlInfoDataSource.cs

```diff
@@ -194,105 +197,70 @@ public async Task<ControlInfoDataGroup> GetGroupFromItemAsync(string uniqueId)
 
         private async Task GetControlInfoDataAsync()
         {
-            lock (_lock)
+            if (this.Groups.Count() != 0)
             {
-                if (this.Groups.Count() != 0)
-                {
-                    return;
-                }
+                return;
             }
 
-            string jsonText = await FileLoader.LoadText("DataModel/ControlInfoData.json");
-
-            JsonObject jsonObject = JsonObject.Parse(jsonText);
-            JsonArray jsonArray = jsonObject["Groups"].GetArray();
-
-            lock (_lock)
+            var jsonFile = await FileLoader.GetJsonFile("DataModel/ControlInfoData.json");
+            using FileStream openStream = File.OpenRead(jsonFile.Path);
+            var controlInfoDataGroup = await JsonSerializer.DeserializeAsync<Root>(openStream, new JsonSerializerOptions
+            {
+                WriteIndented = true,
```

Comment:
Intentional?

Reply:
for using async we need to use stream

Reply:
But do we need `WriteIndented` ?

Reply:
No we dont need for deserializing, Done.